bin/commands/cmd_gen.ml

```diff
@@ -30,7 +30,35 @@ let rec atdgen ?(rescript = false) ?output_prefix:original_prefix = function
     | 0, i | i, _ ->
       Error (`Atd_error (i, file)))
 
-let run ?output_prefix ?(rescript = false) files =
+let rec checkout ~git_file = function
+  | [] ->
+    Ok []
+  | file :: tail ->
+    let open Letops.Result in
+    let* version, commit = Generator.split_commit file in
+    let* filename, _extension = Generator.split_filename git_file in
+    (* This checks out the commit of the versioned file in the same location as the file.
+     * Dangerous?! Should it be in output_prefix, or in a temporary directory?
+     *)
+    let temp_file = filename ^ "_" ^ string_of_int version ^ ".atd" in
+    let commit = match commit with None -> "HEAD" | Some commit -> commit in
+    let cmd = "git show " ^ commit ^ ":" ^ git_file ^ " > " ^ temp_file in
+    (match Sys.command cmd with
+    | 0 ->
+      let* tail = checkout ~git_file tail in
+      Ok (temp_file :: tail)
+    | retval ->
+      Error (`Git_error retval))
+
+let run ?output_prefix ?(rescript = false) ?(git = None) files =
```

Comment:
agreed it should be the other way around, `--files` should be the option

Reply:
Makes sense!
So right now the positional arguments can either be files or commit hashes. I can change the handling of the two cases so that it's git by default. However we still need `--git` or we can treat one of the positional arguments as the filename and the rest as hashes.

Also I'm a little confused by `Cmdliner`. Is there a way to check all positional arguments for "file exists" only if the `--files` option is checked?

Reply:
I don't know, I'll check, but Cmdliner API is quite hard to understand!